components/content_settings/core/browser/brave_content_settings_pref_provider_unittest.cc

```diff
@@ -0,0 +1,330 @@
+  cookie_settings.CheckSettingsWouldBlock(GURL("http://brave.com:5555"));
+  cookie_settings.CheckSettingsWouldBlock(GURL("https://brave.com"));
+  // Check that settings would allow allow.brave.com with any protocol and port.
+  cookie_settings.CheckSettingsWouldAllow(url2);
+  cookie_settings.CheckSettingsWouldAllow(GURL("https://allowed.brave.com"));
+  // Check the pattern that didn't need to be migrated.
+  cookie_settings.CheckSettingsWouldBlock(
+      GURL("https://help.brave.com/article1.html"));
+  // Would not block a different domain.
+  cookie_settings.CheckSettingsAreDefault(GURL("http://brave2.com"));
+
+  // Fingerprinting.
+  // Check that settings would allow brave.com with any protocol and port.
+  fp_settings.CheckSettingsWouldAllow(url);
+  fp_settings.CheckSettingsWouldAllow(GURL("http://brave.com:5555"));
+  fp_settings.CheckSettingsWouldAllow(GURL("https://brave.com"));
+  // Would not allow a different domain.
+  fp_settings.CheckSettingsAreDefault(GURL("http://brave2.com"));
+
+  // HTTPSE.
+  // Check that settings would block brave.com with any protocol and port.
+  httpse_settings.CheckSettingsWouldBlock(url);
+  httpse_settings.CheckSettingsWouldBlock(GURL("http://brave.com:5555"));
+  // Would not block a different domain.
+  httpse_settings.CheckSettingsAreDefault(GURL("http://brave2.com"));
+
+  // Ads.
+  // Check that settings would allow brave.com with any protocol and port.
+  ads_settings.CheckSettingsWouldAllow(url);
+  ads_settings.CheckSettingsWouldAllow(GURL("http://brave.com:5555"));
+  ads_settings.CheckSettingsWouldAllow(GURL("https://brave.com"));
+  // Would not allow a different domain.
+  ads_settings.CheckSettingsAreDefault(GURL("http://brave2.com"));
+
+  // Enabled.
+  // Check that settings would block brave.com with any protocol and port.
+  httpse_settings.CheckSettingsWouldBlock(url);
+  httpse_settings.CheckSettingsWouldBlock(GURL("http://brave.com:5555"));
+  httpse_settings.CheckSettingsWouldBlock(GURL("https://brave.com"));
+  // Would not block a different domain.
+  httpse_settings.CheckSettingsAreDefault(GURL("http://brave2.com"));
+
+  // Scripts.
+  // Check that settings would block brave.com with any protocol and port.
+  script_settings.CheckSettingsWouldBlock(url);
+  script_settings.CheckSettingsWouldBlock(GURL("http://brave.com:5555"));
+  script_settings.CheckSettingsWouldBlock(GURL("https://brave.com"));
+  // Would not block a different domain.
+  script_settings.CheckSettingsAreDefault(GURL("http://brave2.com"));
+
+  // Unknown resource_id - should not have been migrated.
+  // Check that settings would block brave.com:8080, but not brave.com:5555.
+  unknown_resource_id_settings.CheckSettingsWouldBlock(url);
+  unknown_resource_id_settings.CheckSettingsAreDefault(
+      GURL("http://brave.com:5555"));
+
+  provider.ShutdownOnUIThread();
+}
+
+TEST_F(BravePrefProviderTest, TestShieldsSettingsMigrationVersion) {
+  TestingProfile testing_profile;
```

Comment:
this can live in the test fixture

Reply:
Fixed in bdc2851b5386b1ae42a134550a217edb6267ef57